phasespace/fromdecay/genmultidecay.py

```diff
@@ -0,0 +1,290 @@
+import itertools
+from typing import Callable, Union
+
+import tensorflow as tf
+import tensorflow.experimental.numpy as tnp
+from particle import Particle
+
+from phasespace import GenParticle
+
+from .mass_functions import _DEFAULT_CONVERTER
+
+_MASS_WIDTH_TOLERANCE = 0.01
+_DEFAULT_MASS_FUNC = "relbw"
+
+
+class GenMultiDecay:
+    def __init__(self, gen_particles: list[tuple[float, GenParticle]]):
+        """A container that works like GenParticle that can handle multiple decays. Can be created from.
+
+        Args:
+        gen_particles: All the GenParticles and their corresponding probabilities.
+            The list must be of the format [[probability, GenParticle instance], [probability, ...
+        """
+        self.gen_particles = gen_particles
+
+    @classmethod
+    def from_dict(
+        cls,
+        dec_dict: dict,
+        mass_converter: dict[str, Callable] = None,
+        tolerance: float = _MASS_WIDTH_TOLERANCE,
+    ):
+        """Create a GenMultiDecay instance from a dict in the decaylanguage format.
+
+        Args:
+            dec_dict:
+                The input dict from which the GenMultiDecay object will be created from.
+                A dec_dict has the same structure as the dicts used in DecayLanguage.
+            mass_converter: A dict with mass function names and their corresponding mass functions.
+                These functions should take the particle mass and the mass width as inputs
+                and return a mass function that phasespace can understand.
+                This dict will be combined with the predefined mass functions in this package.
+                TODO more docs here
+            tolerance: Minimum mass width of the particle to use a mass function instead of

```

Comment:
You could here insert the default value when building the string, using _MASS_WIDTH_TOLERANCE.

Reply:
Fixed in 0b6ec078e05213984811275d063028b07a477c1d